packages/legend-application/src/components/shared/PackageableElementOptionRenderer.tsx

```diff
@@ -14,23 +14,64 @@
  * limitations under the License.
  */
 
-import type { PackageableElement } from '@finos/legend-graph';
+import type {
+  PackageableElement,
+  GraphManagerState,
+} from '@finos/legend-graph';
 import type { PackageableElementOption } from '../../stores/shared/PackageableElementOption.js';
 
-export const getPackageableElementOptionalFormatter = (props?: {
+export const getPackageableElementOptionFormatter = (props: {
   darkMode?: boolean;
+  graphManagerState: GraphManagerState;
 }): ((
   option: PackageableElementOption<PackageableElement>,
 ) => React.ReactNode) =>
   function PackageableElementOptionLabel(
     option: PackageableElementOption<PackageableElement>,
   ): React.ReactNode {
-    const className = props?.darkMode
+    const className = props.darkMode
       ? 'packageable-element-format-option-label--dark'
       : 'packageable-element-format-option-label';
 
+    const getColor = (element: PackageableElement): string => {
+      switch (true) {
+        case props.graphManagerState.isSystemElement(element):
+          return 'system';
+        case props.graphManagerState.isGeneratedElement(element):
+          return 'generated';
+        case props.graphManagerState.isMainGraphElement(element):
+          return 'main';
+        case props.graphManagerState.isDependencyElement(element):
+          return 'dependency';
+        default:
+          return '';
+      }
+    };
+
+    const getToolTip = (element: PackageableElement): string => {
```

Comment:
Also, these function can be moved outside the body of a component. Generally, we try to keep the body of a function light of logic